.github/workflows/ci.yml

```diff
@@ -9,293 +9,420 @@ on: # rebuild any PRs and main branch changes
+        uses: actions/checkout@v3
         with:
           submodules: recursive
-      - uses: actions-rs/toolchain@v1
+
+      - name: 'Cache index.node'
+        id: cached-artifact
+        uses: actions/cache@v3
         with:
-          toolchain: stable
-      - uses: actions/setup-node@v1
+          path: ./packages/core-bridge/releases
+          key: corebridge-artifactcache-${{ matrix.platform }}-${{ hashFiles('./packages/core-bridge/**/Cargo.lock', './packages/core-bridge/**/*.rs') }}
+
+      - name: Install Rust
+        if: steps.cached-artifact.outputs.cache-hit != 'true'
+        uses: dtolnay/rust-toolchain@stable
         with:
-          node-version: ${{ matrix.node }}
+          target: ${{ matrix.target }}
+
       - name: Install protoc
-        uses: arduino/setup-protoc@v1
+        if: steps.cached-artifact.outputs.cache-hit != 'true'
+        # arduino/setup-protoc@v3 require Node 20, which doesn't work on Ubuntu 18.04 runners
+        uses: arduino/setup-protoc@v2
         with:
-          version: '3.x'
+          # TODO: Upgrade proto once https://github.com/arduino/setup-protoc/issues/99 is fixed
+          version: '23.x'
           repo-token: ${{ secrets.GITHUB_TOKEN }}
-      - uses: Swatinem/rust-cache@v1
-        with:
-          working-directory: packages/core-bridge
-      # Don't build during install phase since we're going to explicitly build
-      - run: npm ci --ignore-scripts
-      - run: npm run build
-        env:
-          BUILD_CORE_RELEASE: true
 
-      # docker-compose
-      - name: Checkout docker-compose repo
-        uses: actions/checkout@v2
-        with:
-          repository: temporalio/docker-compose
-          path: docker-compose
-        if: ${{ startsWith(matrix.os, 'ubuntu') }}
+      - name: Print libc version
+        if: (steps.cached-artifact.outputs.cache-hit != 'true') && startsWith(matrix.platform, 'linux')
+        run: ldd --version
 
-      - name: Start Temporal Server
-        run: docker-compose up -d
-        working-directory: docker-compose
-        if: ${{ startsWith(matrix.os, 'ubuntu') }}
+      - name: Install gcc-aarch64-linux-gnu
+        if: (steps.cached-artifact.outputs.cache-hit != 'true') && (matrix.platform == 'linux-arm')
+        run: sudo apt update && sudo apt install -y gcc-aarch64-linux-gnu
 
-      - name: Wait for Temporal Server
-        run: npm run wait-namespace
-        if: ${{ startsWith(matrix.os, 'ubuntu') }}
-      # end:docker-compose
+      - name: Rust Cargo and Build cache
+        if: steps.cached-artifact.outputs.cache-hit != 'true'
+        # Starting with 2.7.2, rust-cache requires Node 20, which doesn't work on Ubuntu 18.04 runners
+        uses: Swatinem/rust-cache@v2.7.1
+        with:
+          workspaces: packages/core-bridge -> target
+          prefix-key: corebridge-buildcache
+          shared-key: ${{ matrix.platform }}
+          env-vars: ''
+          # save-if: ${{ env.IS_MAIN_BRANCH }}
```

Comment:
No. I will uncoment this at the very last moment. Added a FIXME.

Cache space is limited, and rust build cache can take a lot of space. I want to avoid cache trashing due to PRs that make several successive modifications to the native code.
So the idea here is to keep the rust build cache reflecting the main branch as much as possible.